src/coreclr/jit/lower.cpp

```diff
@@ -6792,6 +6792,40 @@ void Lowering::LowerStoreIndirCommon(GenTreeStoreInd* ind)
     TryCreateAddrMode(ind->Addr(), true, ind->TypeGet());
     if (!comp->codeGen->gcInfo.gcIsWriteBarrierStoreIndNode(ind))
     {
+        if (varTypeIsFloating(ind) && ind->Data()->IsCnsFltOrDbl())
+        {
+            // Optimize *x = DCON to *x = ICON which is slightly faster on xarch
+            GenTree*  data   = ind->Data();
+            double    dblCns = data->AsDblCon()->gtDconVal;
+            ssize_t   intCns = 0;
+            var_types type   = TYP_UNKNOWN;
+
+            if (ind->TypeIs(TYP_FLOAT)
+#if defined(TARGET_ARM64) || defined(TARGET_ARM)
+                && data->IsFPZero()
+#endif
+            )
+            {
+                float fltCns = static_cast<float>(dblCns); // should be a safe round-trip
+                intCns       = static_cast<ssize_t>(*reinterpret_cast<INT32*>(&fltCns));
+                type         = TYP_INT;
+            }
+#ifdef TARGET_AMD64
```

Comment:
Switch to `TARGET_64BIT` to contain zero doubles for ARM64 as well?

Reply:
> Switch to TARGET_64BIT to contain zero doubles for ARM64 as well?

Yes, corrected